src/f4enix/input/MCNPinput.py

```diff
@@ -229,6 +261,50 @@ def __init__(
         # cells.extend(surfs)
         # cells.extend(data)
 
+    @property
+    def cells(self) -> dict[str, parser.Card]:
+        """Get the cells attribute."""
+        return self._cells
+
+    @cells.setter
+    def cells(self, value: dict[str, parser.Card]) -> None:
+        """Set the cells attribute."""
+        # enforce that the keys are all corresponding to the cell number
+        # in case of difference, the dictionary key takes precedence
+        for key, card in value.items():
+            if key != str(card.name):
+                logging.warning(
+                    f"The cell number {card.name} is different from Key {key} and will be replaced"
+                )
+                card.name = int(key)
+                card._set_value_by_type("cel", int(key))
```

Comment:
Add a check that get_values was run before using _set_value_by_type (even though very rarely it is not used before)

Reply:
In F4Enix the get_values() is called on all cards always. 
The scenario you are presenting may happen if someone is creating a card from scratch using directly numjuggler. I tend to think that in that case it is its responsability to call the get_value() method on it before adding to the F4Enix input.

What do you think about that? is there any other scenario where this could happen?